### PR TITLE
Ongoing work for expenses refactoring

### PR DESCRIPTION
**Pull request by sedubois**
Can be merged as-is and resumed later.

### CODE REVIEW

**Comment by xdamman** [comment]
Since it's not urgent I'd love to wait for @asood123 to be back to review this.

File: server/models/Expense.js [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+
+      approve() {
+        this.status = status.APPROVED;
+        return this.save();
+      }
+    }
+  });
+
+  return Expense;
+  // return Temporal(Expense, Sequelize);
```
Comment (asood123) [inline server/models/Expense.js]: Can you make this a TODO: enable Temporal, so all expense changes are recorded?

File: server/models/index.js [hunk 2]
```diff
@@ -78,6 +79,8 @@ function setupModels(client) {
   m.User.hasMany(m.Activity);
 
   m.Activity.belongsTo(m.Transaction);
+  m.Activity.belongsTo(m.Expense);
```
Comment (asood123) [inline server/models/index.js]: I don't think we need to add `Expense` field in `Activity` model. Would rather keep everything in the `data` field.

File: server/models/index.js [hunk 3]
```diff
@@ -78,6 +79,8 @@ function setupModels(client) {
...
+  // TODO shouldn't there also be a donation-activity relation?
```
Comment (asood123) [inline server/models/index.js]: Same here. Don't need Donation in Activity table either.

**Comment by asood123** [comment]
Overall looks good. Few minor fixes and will be ready to be merged.

**Comment by asood123** [comment]
@sedubois can we wrap this up this week? It shouldn't take more than a few mins

**Comment by sedubois** [comment]
@asood123 review rdy

**Pull request merged**